src/GraphQL/Resolver.hs

```diff
@@ -241,92 +254,133 @@ resolveField handler nextHandler field =
     let argName = getName argument
     value <- case lookupArgument field argName of
       Nothing -> valueMissing @t argName
       Just v -> first (InvalidValue argName) (fromValue @t v)
     buildFieldResolver @m @f (handler value) field
 
+instance forall ksK t f m name.
+  ( KnownSymbol ksK
+  , BuildFieldResolver m f
+  , KnownSymbol name
+  , Defaultable t
+  , API.GraphQLEnum t
+  , Monad m
+  ) => BuildFieldResolver m (EnumField (API.Argument ksK (API.Enum name t)) f) where
+  buildFieldResolver handler field = do
+    argName <- first SchemaError (AST.nameFromSymbol @ksK)
+    value <- case lookupArgument field argName of
+      Nothing -> valueMissing @t argName
+      Just (ValueEnum enum) -> first (InvalidValue argName) (API.enumFromValue @t enum)
+      Just value -> Left (InvalidValue argName (show value <> " not an enum: " <> show (API.enumValues @t)))
+    buildFieldResolver @m @f (handler value) field
+
+-- Note that we enumerate all ks variables with capital letters so we
+-- can figure out error messages like the following that don't come
+-- with line numbers:
+--
+--        • No instance for (GHC.TypeLits.KnownSymbol ks0)
+--            arising from a use of ‘interpretAnonymousQuery’
 
 -- We only allow Field and Argument :> Field combinations:
 type family RunFieldsType (m :: Type -> Type) (a :: [Type]) = (r :: Type) where
-  RunFieldsType m '[API.Field ks t] = API.Field ks t
+  RunFieldsType m '[API.Field ksI t] = API.Field ksI t
   RunFieldsType m '[a :> b] = a :> b
-  RunFieldsType m ((API.Field ks t) ': rest) = API.Field ks t :<> RunFieldsType m rest
+  RunFieldsType m ((API.Field ksJ t) ': rest) = API.Field ksJ t :<> RunFieldsType m rest
   RunFieldsType m ((a :> b) ': rest) = (a :> b) :<> RunFieldsType m rest
   RunFieldsType m a = TypeError (
     'Text "All field entries in an Object must be Field or Argument :> Field. Got: " ':<>: 'ShowType a)
 
 -- Match the three possible cases for Fields (see also RunFieldsType)
 type family RunFieldsHandler (m :: Type -> Type) (a :: Type) = (r :: Type) where
-  RunFieldsHandler m (f :<> fs) = FieldHandler m f :<> RunFieldsHandler m fs
-  RunFieldsHandler m (API.Field ks t) = FieldHandler m (API.Field ks t)
-  RunFieldsHandler m (a :> b) = FieldHandler m (a :> b)
+  RunFieldsHandler m (f :<> fs) = FieldHandler m (FieldResolverDispatchType f) :<> RunFieldsHandler m fs
+  RunFieldsHandler m (API.Field ksL t) = FieldHandler m (FieldResolverDispatchType (API.Field ksL t))
+  RunFieldsHandler m (a :> b) = FieldHandler m (FieldResolverDispatchType (a :> b))
   RunFieldsHandler m a = TypeError (
     'Text "Unexpected RunFieldsHandler types: " ':<>: 'ShowType a)
 
 
 class RunFields m a where
-  -- runFields is run on a single AST.Selection so it can only ever
-  -- return one ObjectField.
+  -- runFields runs a a single AST.Selection over all possible fields
+  -- (as specified by the type), returing exactly one ObjectField when
```

Comment:
Fixed. Also, made this a doc comment (`-- |` rather than `--`) and made some markup & wording tweaks.